src/pydisagg/DisaggModel.py

```diff
@@ -21,574 +21,677 @@ class DisaggModel:
-            the age density of the population in the form of an array (grid of densities)
-        '''
-        return np.sum(bucket_populations*self.predict_rates(beta))
-
-    def _rate_derivative(self, beta, rate_pattern_value):
-        '''
-        Computes the derivative with respect to beta
-        of the predicted rate in a bucket with a fixed rate_pattern value
-        '''
-        return 1/self.T_diff(self.T_inverse(beta + self.T(rate_pattern_value)))
-
-    def _H_diff(self, beta, bucket_populations):
-        '''
-        Function outputs the derivative of the above _H_func with respect to beta
-        '''
-        return np.sum(bucket_populations *
-                      self._rate_derivative(beta, self.rate_pattern)
-                      )
+        return self.predict_rate(beta,rate_pattern)*bucket_populations
+    
+    def count_diff_beta(
+        self,
+        beta: float,
+        rate_pattern: NDArray,
+        bucket_populations: NDArray,
+    )->NDArray:
+        """
+        Computes the derivative of the predicted count in each bucket with respect to beta
 
+        Parameters
+        ----------
+        beta : float
+            beta paramter, log of rescaling parameter
+        rate_pattern : NDArray
+            rate pattern 
+        bucket_populations :NDArray
+            Populations in each bucket
+        Returns
+        -------
+        NDArray
+            Derivative of predicted counts in each bucket with respect to beta
+        """
+        return self.rate_diff_beta(beta,rate_pattern)*bucket_populations
+    
     def fit_beta(
         self,
-        bucket_populations: NDArray,
         observed_total: float,
-        observed_total_se: Optional[float] = None,
-        rate_pattern: Optional[NDArray] = None,
+        rate_pattern: NDArray,
+        bucket_populations: NDArray,
         lower_guess: Optional[float] = -50,
         upper_guess: Optional[float] = 50,
         verbose: Optional[int] = 0
-    )->None:
-        """
-        Fit a value for beta from the age density of a population and a measured count
-        Will attempt to generate a standard error using delta method if a standard error for
-        for age density is given.
+    ) -> float:
+        """Fits the parameter beta to the data
 
         Parameters
         ----------
-        bucket_populations : NDArray
-            numpy array of population sizes for each bucket
         observed_total : float
-            total of observed quantity across bucket_populations
-        observed_total_se : Optional[float], optional
-            se of observed_total, by default None
-        rate_pattern : Optional[NDArray], optional
-            , by default None
+            aggregated observation
+        rate_pattern : NDArray
+            rate pattern to assume (generalized) proportionality to
+        bucket_populations : NDArray
+            populations in each bucket
         lower_guess : Optional[float], optional
```

Comment:
I would use `lower_guess : float, optional` instead

Reply:
Thanks for pointing this out, fixed in commit acb2254ae7d4ef4fdcb10662c6dcd656110c35ce